test/test_regression_cli.py

```diff
@@ -39,116 +59,69 @@ def setup_module():
-        os.system("retriever install sqlite %s -f output_database" % dataset)
-        os.system("echo .dump | sqlite3 output_database > output_file")
-        current_md5 = getmd5(unixfileformat("output_file"))
-        assert current_md5 == known_md5
-
-
-class CSVRegression(TestCase):
-
-    def check_csv_regression(self, dataset, known_md5):
-        """Check for regression for a particular dataset imported to csv"""
-        os.system("rm output_file*")
-        os.system("retriever install csv %s -t output_file_{table}" % dataset)
-        os.system("cat output_file_* > output_file")
-        current_md5 = getmd5('output_file')
-        assert current_md5 == known_md5
-
-
-class MySQLRegression(TestCase):
-
-    def check_mysql_regression(self, dataset, known_md5):
-        """Check for regression for a particular dataset imported to mysql"""
-        os.system("rm output_file*")
-        os.system('mysql -u travis -Bse "DROP DATABASE IF EXISTS testdb"') # installing over an existing database changes the dump
-        os.system("retriever install mysql %s -u travis -d testdb" %dataset)  # user 'travis' for Travis CI
-        os.system("mysqldump testdb -u travis --compact --compatible=no_table_options --no-create-db --no-create-info --result-file=output_file")
-        current_md5 = getmd5(unixfileformat("output_file"))
-        assert current_md5 == known_md5
-
-
-class PostgreSQLRegression(TestCase):
-
-    def check_postgres_regression(self, dataset, known_md5):
-        """Check for regression for a particular dataset imported to postgres"""
-        os.system("rm output_file*")
-        os.system('psql -U postgres -d testdb -h localhost -c "DROP SCHEMA IF EXISTS testschema CASCADE"')
-        os.system("retriever install postgres %s -u postgres -d testdb -a testschema" % dataset)
-        os.system("pg_dump -n testschema --data-only -U postgres -h localhost -f output_file testdb")
-        current_md5 = getmd5(unixfileformat("output_file"))
-        assert current_md5 == known_md5
-
-
-class DownloadRegression(TestCase):
-
-    def check_download_regression(self, dataset, known_md5):
-        """Check for regression for a particular dataset downloaded only"""
-        os.system("retriever download {0} -p raw_data/{0}".format(dataset))
-        current_md5 = getmd5("raw_data/%s" % (dataset))
-        assert current_md5 == known_md5
-
-
-dbms_test_classes = {'sqlite': SqliteRegression, 'csv': CSVRegression,
-                     'mysql': MySQLRegression, 'postgres': PostgreSQLRegression,
-                     'download': DownloadRegression}
-
-for dbms in known_md5s:
-    for dataset in known_md5s[dbms]:
-        stub_test = _test_factory('check_%s_regression' % dbms, 'test_%s' % dataset,
-                                  dataset, known_md5s[dbms][dataset])
-        setattr(dbms_test_classes[dbms], stub_test.__name__, stub_test)
-        del(stub_test)
-
-if __name__ == '__main__':
-    nose.runmodule()
+def dump_dir():
```

Comment:
Can this code be moved to pytest's `setup_function()` and then the explicit execution of `dump_dir` removed from line 44?